src/main/java/org/jenkinsci/plugins/saml/SamlEncryptionData.java

```diff
@@ -29,15 +30,15 @@
  */
 public class SamlEncryptionData {
     private final String keystorePath;
-    private final String keystorePassword;
-    private final String privateKeyPassword;
+    private final Secret keystorePassword;
+    private final Secret privateKeyPassword;
```

Comment:
Leave the original fields there, but `@Deprecated transient`; introduce new fields, migrated with `readResolve`, nulling out the old ones.